site/src/modules/dashboard/Navbar/NavbarView.tsx

```diff
@@ -342,57 +342,58 @@ const ProxyMenu: FC<ProxyMenuProps> = ({ proxyContextValue }) => {
 
         <Divider css={{ borderColor: theme.palette.divider }} />
 
-        {proxyContextValue.proxies
-          ?.sort((a, b) => {
-            const latencyA = latencies?.[a.id]?.latencyMS ?? Infinity;
-            const latencyB = latencies?.[b.id]?.latencyMS ?? Infinity;
-            return latencyA - latencyB;
-          })
-          .map((proxy) => (
-            <MenuItem
-              key={proxy.id}
-              selected={proxy.id === selectedProxy?.id}
-              css={{ fontSize: 14 }}
-              onClick={() => {
-                if (!proxy.healthy) {
-                  displayError("Please select a healthy workspace proxy.");
+        {proxyContextValue.proxies &&
+          [...proxyContextValue.proxies]
```

Comment:
This is the only major change – check that `proxies` is defined, and then if it is, spread it into a new array, and mutate that instead of the original